pkg/cloud/google/machineactuator.go

```diff
@@ -796,19 +800,51 @@ func getOrNewComputeServiceForMachine(params MachineActuatorParams) (GCEClientCo
 	if params.ComputeService != nil {
 		return params.ComputeService, nil
 	}
-	// The default GCP client expects the environment variable
-	// GOOGLE_APPLICATION_CREDENTIALS to point to a file with service credentials.
-	client, err := google.DefaultClient(context.TODO(), compute.ComputeScope)
-	if err != nil {
-		return nil, err
+
+	var client *http.Client
+	var err error
+	// If specified in the GCE config, use the alternative authentication.
+	if params.CloudConfigPath != "" {
+		glog.Info("Trying to get open the GCE config")
+		client, err = clientWithAltTokenSource(params.CloudConfigPath)
 	}
+	if params.CloudConfigPath == "" || err != nil {
```

Comment:
Would it be better to crash if an invalid config file is provided instead of falling back to the default client? If a file is specified and ends up not being used (due to a parsing error) that seems like a confusing situation to debug.

Reply:
Makes sense. Changed.